tests/network/probes.go

```diff
@@ -319,3 +353,31 @@ func getVMIConditions(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineI
 	Expect(err).ToNot(HaveOccurred())
 	return readVmi.Status.Conditions
 }
+
+func newRandomFedoraVMIWithGuestAgent(opts ...libvmi.Option) *v1.VirtualMachineInstance {
+	networkData, err := libnet.CreateDefaultCloudInitNetworkData()
+	Expect(err).NotTo(HaveOccurred())
+
+	vmi := libvmi.NewFedora(
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+		libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		libvmi.WithCloudInitNoCloudNetworkData(networkData, false),
+	)
+
+	for _, opt := range opts {
+		opt(vmi)
+	}
```

Comment:
Can't we just add `opts` support to `NewRandomFedoraVMIWithGuestAgent` in tests/utils.go? (the function should probably be moved to libvmi but that's another topic...)
This function creates duplication by creating yet another fedora VMI function, and doesn't seem to include anything specific to network probes.

Reply:
@maiqueb I may not have been understood.
The build pattern can be used here, I actually strongly encourage to use only it allover for both consistency and readability.
 
 I think @AlonaKaplan found a good balance [here](https://github.com/kubevirt/kubevirt/pull/7520/commits/210f8ea964cf50825b9cb7b9b25958edce4bb9f6#diff-eda903ca203c0ef35a6eaa4bff5ad43ef1eecfc83ea7d135d3b925f11a4b6406R46).

If you can build groups of options that fit your need, do that over creating a factory.
Sometime, like in the example from the ref PR, a portion of the options are widely used and theregore merit adding in libvmi, sometime it is less used and these can be added at specific domains.
But you can always keep using the build pattern.

BTW, the fedora image has the guest-agent included, adding “random” and “with guest agent” wording is redundant. But again, the factory is not needed at all IMO.

Reply:
Neat move. Thanks for raising my attention towards it.